predicators/structs.py

```diff
@@ -1808,6 +1808,18 @@ def __len__(self) -> int:
         return len(self.ground_nsrts)
 
 
+@dataclass(frozen=True, repr=False, eq=False)
+class SpotActionExtraInfo:
+    """A sequence of things that are in the extra_info field of actions for the
+    SpotEnv."""
+    action_name: str
+    operator_objects: Sequence[Object]
+    real_world_fn: Optional[Callable]
+    real_world_fn_args: Sequence
```

Comment:
should we make this a Tuple instead? same for below. for example, you're needing to assert that it's an instance of a tuple in at least one place. if you change the type, you should be able to remove that assertion